### ISSUE TITLE
Update Emoji library

### ISSUE THREAD

**Issue by matthias4217** [RocketChat/Rocket.Chat#24917]
### Description:

Currently, emojis in the emoji picker are limited to Emoji 12. The current set seem to be the one in the npm package `emojione` (`emojione-assets` in the dev dependencies). However, this project has been discontinued due to a change of name. It should be replaced by the newer `emoji-toolkit` package. See more about it in [this Readme](https://github.com/joypixels/emoji-toolkit/blob/master/UPGRADE.md).

This is of course a minor issue, but I figure it can also be quite easy to fix. Thank you for your time and attention !

### Steps to reproduce:

Search for an Emoji 12 or newer emoji, such as 🪓.

### Expected behavior:

I would like to have a more up-to-date version.

### Actual behavior:

There is no 🪓 in the emoji picker.

### Server Setup Information:

- Version of Rocket.Chat Server: `4.5.1`
- Operating System: Varies
- Deployment Method: Docker
- Number of Running Instances: 4
- DB Replicaset Oplog: 
- NodeJS Version: v14.18.3
- MongoDB Version:5.0.3

### Client Setup Information

- Desktop App or Browser Version: Both
- Operating System: Ubuntu 20.04

### Update

As of RocketChat 6.13.0, the issue still stands. Unicode 16.0 has been released on September 10<sup>th</sup>, 2025. RocketChat emojis are missing about seven years of updates.

**Comment by mpgent** [comment]
I'd like to work on this issue

**Comment by ankar84** [comment]
Nice to see new updated emojis in Rocket Chat!

**Comment by Zishan-7** [comment]
Hey! I would love to work on this issue.

**Comment by meerhamzadev** [comment]
Is this issue still open? I would like to give it a try.

**Comment by matthias4217** [comment]
This issue is still opened indeed. Feel free to make a pull request I guess !

**Comment by Illusion47586** [comment]
Hello @matthias4217 I would like to work on this issue

**Comment by aliubaidsaifi** [comment]
Hello @matthias4217 I would like to work on this issue please assign it to me.

**Comment by matthias4217** [comment]
> Hello @matthias4217 I would like to work on this issue please assign it to me.

Hello, I am not a RocketChat maintainer. Feel free to contribute. @kotwani2883 has already opened the PR #25131 but it hasn't been reviewed yet.

**Comment by adityajangir** [comment]
Hello @matthias4217 , Could you please assign this issue to me.

**Comment by matthias4217** [comment]
> Hello @matthias4217 , Could you please assign this issue to me.

Hello, I am not a RocketChat maintainer. I cannot assign anyone to it, you're absolutely free to open a Pull Request to work on this issue !

**Comment by anikdhabal** [comment]
I would like to work on this issue.please assign this issue to me.

**Comment by kartikbh6614** [comment]
I want to work on this issues please assign me this issue .

**Comment by 21SINGH** [comment]
this is my first time here i pushed a commit request , but its showing review required , i dont understand a thing what's going on can anyone please explaine its like that since yesterday

**Comment by anikdhabal** [comment]
@21SINGH  your code required  review before merge into the project. Reviews allow collaborators to comment on the changes  in pull your requests, approve the changes, or request further changes before the pull request is merged. Repository administrators can require that all pull requests are approved before being merged.

**Comment by 21SINGH** [comment]
so then who will review it 
if the administrator is going to do it , then when will he do it 
because i posted it yesterday

**Comment by seniorLikeToCode** [comment]
hi I want to contribute to this , assign me this issue.

**Comment by LovedeepSingh19** [comment]
hi I need some help with this issue. When ever i try to update the import emojione to emoji-toolkit it shows no library found. I try to install it with meteor and it get installed but in the end whenever i try to run yarn dev it just keep on crashing the application can anyone help me with this  one plz.

**Comment by MashiatK** [comment]
hi I want to contribute to this , assign me this issue please.

**Comment by VaibhavAPatil** [comment]
I am **interested** in **this Issue**. Can I do **Pull Request** or you can **assign** this **issue to Me**.

**Comment by nikhil-babar** [comment]
I am interested in solving this issue and can add this new dependency. Please can you assign me this issue.

**Comment by dsg1320** [comment]
Hey can you please assign this issue

**Comment by sachinrastogi12** [comment]
Hey please assign it to me, I want to contribute.

**Comment by B1TSH3LL** [comment]
if the issue isn't solved yet, please assign this to me. i would work on this issue.

**Comment by thekrprince** [comment]
Hi, Please assign this issue to me. I want to contribute.

**Comment by Jashwanthkanderi** [comment]
does anyone working on this issue cause I can contribute to it @matthias4217

**Comment by matthias4217** [comment]
@Jashwanthkanderi There’s a merge request opened here (https://github.com/RocketChat/Rocket.Chat/pull/27782) but it hasn’t been accepted yet.

**Comment by Jashwanthkanderi** [comment]
> @Jashwanthkanderi There’s a merge request opened here (https://github.com/RocketChat/Rocket.Chat/pull/27782) but it hasn’t been accepted yet.

Yeah i don't know why it's been 10+ hrs I think it should be correct , but I may solve this issue

**Comment by KshitjMaheshwari** [comment]
Hi Please assign this to me. I want to contribute.

**Comment by adastra-react** [comment]
Hey, I would really like to work on this issue, this is my first time getting into open source.

**Comment by ranjanmangla1** [comment]
Hey, if this issue is still open, i would really love to work on this issue. Kindly assign me this issue.

**Comment by matthias4217** [comment]
@ranjanmangla1 You don’t need to be assigned on it to work on this issue. [This comment](https://github.com/RocketChat/Rocket.Chat/pull/27799#issuecomment-1520473765) also gives some advice. ^^

**Comment by auspy** [comment]
starting working on this issue

**Comment by SyedAbuBakerAli** [comment]
Hi, @matthias4217 is this issue still open I want to work on this issue.
Can you please assign this to me?

**Comment by nick2432** [comment]
can i work on this?

**Comment by devr77** [comment]
@dudanogueira @matthias4217 Is this issue solved, Please Reply

**Comment by SwapnilSonker** [comment]
is this issue fixed by #27782, if not then I want to fix it.

**Comment by Asrani-Aman** [comment]
@matthias4217 is this issue fixed? or else I can work on it?

**Comment by himani-mehra** [comment]
Hi @matthias4217 , I would love to take on this issue and contribute to the project. Could you please assign it to me if it's available? Looking forward to your response. Thanks!

**Comment by RahulJOD** [comment]
Hi, Please assign this issue to me. I want to contribute.

**Comment by nikkivaddepelli** [comment]
I would like to work on this issue. Can you please assign this to me?

**Comment by akuthotaaravind** [comment]
Hey i want to work on this .Do any one working on the isuue let me know @dudanogueira @ankar84 @matthias4217 @mpgent @thekrprince

**Comment by Shriyansh-20** [comment]
@dudanogueira @mpgent @ankar84 Interested, Please assign

**Comment by Harsh-Builds** [comment]
Hey, if this issue is still open, I would really love to work on this issue. Kindly assign me this issue.

**Comment by IayushCoderJOD** [comment]
Is this still open @matthias4217

**Comment by Bhavna1312** [comment]
Hi, I’m a beginner and would like to work on this issue. Can someone guide me on how to get started?

**Comment by harsh41414141414** [comment]
"Hi! I’d love to work on this issue. Can I take it?"

**Comment by MrPrinceSaxena** [comment]
"Hey, I'm a beginner and would like to work on this issue."

**Comment by Niteshprimedev** [comment]
Hi team! 

I came across this issue and it looks like a very interesting (and beginner-friendly) enhancement to work on.

Before diving in:
- Is this issue still open for contributions?
- Has anyone started working on replacing `emojione-assets` with `emoji-toolkit` (or an alternative)?
- If not, I’d love to pick this up and start exploring a fix.

Thanks a lot!

**Comment by Nikita-singh-14** [comment]
Hello,
is this issue still open? If yes, then assign to me. I'd love to pick this up and start exploring a fix.

**Comment by akshatajmera7** [comment]
hey, if this issue is not being resolved. Can i work on this. I wanna contribute to open source

**Comment by Anjali-yadav513** [comment]
"Hi, I'm new to Rocket.Chat open source. I'd like to work on this issue. could you please assign it to me?"

**Comment by Abhrajit-debnath** [comment]
Hi! I would like to work on this issue. Could you please assign it to me?

**Comment by naitikk31** [comment]
Hi, I'm new to Rocket.Chat open source. I'd like to work on this issue. could you  assign it to me?

**Comment by SuyashJain17** [comment]
Hi! I’m interested in working on this.
Can someone confirm if this issue is still valid on the current develop branch?

**Comment by ankit1337-git** [comment]
I would like to work on this issue. Is it still available? Could you please assign it to me?

**Comment by Soham-0816** [comment]
Hi, I’m interested in working on this issue.

Before starting, I’d like to confirm a few things:
• Is replacing emojione-assets with emoji-toolkit the expected approach?
• Are there any compatibility concerns (e.g., older clients or emoji shortcodes)?
• Should the update include full Unicode 16.0 support?

I’ve already set up Rocket.Chat locally and explored the emoji picker code.

It would be great if this issue could be assigned to me. Thanks!

**Comment by Pragati-cse** [comment]
Hi mentors, I would like to work on this issue.
thankyou!

**Comment by absharabi** [comment]
Hi! I'm a new contributor and would like to work on this issue.
Could you please assign it to me? Thanks!

**Comment by Yashika-code** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me?

**Comment by wineerteam** [comment]
Hi team,
I have gone through the issue and the discussion.
I would like to work on updating the emoji library.
Could you please assign this issue to me or let me know if I can proceed?
Thanks!

**Comment by akanshagupta008** [comment]
Hi! I’m new to the Rocket.Chat community and would like to work on this issue.
Please assign it to me. Thank you!

**Comment by CodeMatrix1** [comment]
Hello to whomsoever is concerned with maintaining this issue, I hope to get some clarification as I find that this is a bigger fix than expected as package 'emoji-toolikit' does'nt have images which is required in the pipeline for generating sprites. While this issue could be resolved by externally supplementing images, for example, from github of twemoji, I am not sure contributors are allowed to do asset creation and change the sprite files that are already generated, so I hope somebody can guide me out.

**Comment by hassarch** [comment]
Hello,
I have gone through the issue and the discussion.
I would like to work on this emoji library issue.
Could you please assign this issue to me?
regards.

**Comment by SamathaReddy-Web** [comment]
Hello!  
I'm Samatha, a student developer interested in contributing to Rocket.Chat.  
I went through this issue and would love to work on it.  
Could you please assign it to me?

Thank you!

**Comment by prem-thatikonda29** [comment]
Hi everyone, I've been researching the sprite/image blocker that was raised in the recent comment, and I think I've found a clear path forward.
The core problem is that emoji-toolkit provides metadata but no PNG assets for sprite generation. I looked into compatible image sources and here's what I found:
- Noto Emoji (Google) : image resources are under Apache 2.0, which is fully compatible with Rocket.Chat's MIT license. It covers Unicode 16.0 (the latest standard), is actively maintained by Google Fonts, and PNG assets are available directly from their GitHub repo at googlefonts/noto-emoji. No attribution wall, no commercial restrictions.
- Twemoji : CC BY 4.0, which is also compatible, but it only covers Unicode 14.0 and appears to no longer be actively maintained since Twitter/X's restructuring.

My proposal would be: use emoji-toolkit for metadata (as started in PR #37716) + Noto Emoji PNGs as the image source for sprite generation, replacing emojione-assets entirely. This would bring Rocket.Chat to Unicode 16.0 coverage, about 7 years of missing emojis.
Before proceeding, I'd love a maintainer's input on: (1) whether asset replacement is in scope, and (2) whether Noto Emoji's visual style aligns with the team's design preferences. Happy to take this on once there's a green light.

**Comment by CodeMatrix1** [comment]
hey @prem-thatikonda29 I been working on this issue for over a month till now, I too thought that the approach would be the same as you mentioned but I ended up wasting plenty of time that I could use for other issues if you can see I had commented about it jan 4th

**Comment by CodeMatrix1** [comment]
I am actually done and am opening up a PR in a few minutes, I am really sorry for your 
dissapointment of not being able to work on this issue

**Comment by NETIZEN-11** [comment]
Hi 👋 I would like to work on this issue. 
Can I be assigned?

**Comment by priyavarshini27** [comment]
Hello I'm a GSOC 2026 aspirant with experience in mern stack. I would like to work on this issue

**Comment by engelgabriel** [comment]
Closing this ISSUE in favor of https://github.com/RocketChat/Rocket.Chat/pull/39411, where we’re replacing Emojione with native Unicode emojis.

It’s still a draft, and we’d love help from the community to get it across the finish line 🙌

**Comment by CodeMatrix1** [comment]
hello @engelgabriel , I had made the last standing PR that wd have closed this issue, spent a lot of time (over a month) figuring it out.

If its possible, I’d be happy to contribute to any kind, follow-up tasks, edge cases, or related issues I can help with

 please lmk where exactly i can help :)
